From 1c92ea4f0e7032cec3e19066daf087905737ced0 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 09 May 2022 18:40:09 +0300
Subject: [PATCH] [SW-85036] rom: add efuse prog setting

In gaudi3 as a part of PSOC_EFUSE, a new field PROG_ADDRESS is added
which specifies the write range to efuse. This field is explicitly
set to the default value in soconline as the boot manager can
possibly change this.

This change is recommended by the psoc design team.

Change-Id: I0a98e7f76819d510f3da7eb374aea77f675fea15
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 7b982be..e2eb7e0 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -927,3 +927,8 @@
 			return PCI_IDS_GAUDI3;
 	}
 }
+
+void fuse_prog_range_config(void)
+{
+	regmap_writel(D0_PSOC_DFT_EFUSE_PROG_ADDRESS_LAST_M, mmD0_PSOC_EFUSE_PROG_ADDRESS);
+}
diff --git a/common/efuse.c b/common/efuse.c
index 49475a9..e470018 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -18,6 +18,7 @@
 #define FUSE_DELAY_US		20
 #define FUSE_RETRIES		(FUSE_TIMEOUT_US / FUSE_DELAY_US)
 
+__weak void fuse_prog_range_config(void) {}
 /*
  * fuse box is devided into half for redundancy
  * each half is bank of FUSE_WORDS_PER_BANK words
@@ -38,6 +39,8 @@
 	 */
 	regmap_writel(EFUSE_READ_RANGE_ALL, mmEFUSE_READ_ADDRESS);
 
+	fuse_prog_range_config();
+
 	if (arcboot_rom_global.efuse_read == 1)
 		return -1;
 
